nck/readers/mytarget_reader.py

```diff
@@ -137,12 +136,21 @@ def __get_all_results(self, name_content: str, offset=0) -> List[Dict[str, str]]
         count = first_elements["count"]
         elements = [first_elements["items"]]
         if count > LIMIT_REQUEST_MYTARGET:
-            elements += [
-                self.__get_response(name_content, offset=offset)["items"]
-                for offset in range(
-                    LIMIT_REQUEST_MYTARGET, self.round_up_to_base(count, LIMIT_REQUEST_MYTARGET), LIMIT_REQUEST_MYTARGET
-                )
-            ]
+            range_offset = range(
+                LIMIT_REQUEST_MYTARGET,
+                self.round_up_to_base(count, LIMIT_REQUEST_MYTARGET),
+                LIMIT_REQUEST_MYTARGET
+            )
+            for offset in range_offset:
+                resp = self.__get_response(name_content, offset=offset)
+                count = 0
+                while 'items' not in resp.keys() and count < 10:
+                    resp = self.__get_response(name_content, offset=offset)
+                    count += 1
+                if 'items' in resp.keys():
+                    elements.append(resp['items'])
+                else:
+                    logging.warning('Incorrect response from the API' + str(resp))
```

Comment:
Don't you want to use something like the **"retry"** decorator used in:
-dcm_client
-facebook_reader
-dbm_reader
...
?

Reply:
for example [here](https://github.com/artefactory/nautilus-connectors-kit/blob/630ba9ce882391c2d722e535e6b92c8fb4c2bfa7/nck/readers/dbm_reader.py#L153-L163) in dbm_reader